File: apps/web/types/analysis.ts
// 분석 결과 관련 타입

import type { AnalysisStatus } from './api'

// 디자인 토큰
export interface ColorToken {
  id: string
  name: string
  value: string
  rawValue: { r: number; g: number; b: number; a: number }
  usageCount: number
  cssVariable: string
  tailwindClass: string
  usageNodes: string[]
}

export interface TypographyToken {
  id: string
  name: string
  fontFamily: string
  fontSize: number
  fontWeight: number
  lineHeight: number
  letterSpacing?: number
  tailwindClasses: string
  usageNodes: string[]
}

export interface SpacingToken {
  id: string
  value: number
  tailwindClass: string
  usageContext: string
}

export interface RadiusToken {
  id: string
  value: number
  tailwindClass: string
  usageContext: string
}

export interface DesignTokens {
  colors: ColorToken[]
  typography: TypographyToken[]
  spacing: SpacingToken[]
  radius: RadiusToken[]
}

// 컴포넌트 후보
export interface ComponentCandidate {
  nodeId: string
  figmaName: string
  suggestedName: string
  componentType: 'section' | 'card' | 'ui' | 'layout' | 'unknown'
  filePath?: string
  isRepeating: boolean
  confidence: number
  reasoning: string
  children?: ComponentCandidate[]
  repeatCount?: number
}

export interface Warning {
  type: 'LOW_CONFIDENCE' | 'COMPLEX_LAYOUT' | 'ABSOLUTE_POSITION' | 'MISSING_AUTOLAYOUT'
  nodeId: string
  message: string
}

export interface AiInterpretation {
  componentCandidates: ComponentCandidate[]
  layoutPattern: string
  topLevelSections: string[]
  warnings: Warning[]
  modelUsed: string
  processedAt: string
}

// 코드 생성 결과
export interface GeneratedFile {
  path: string
  type: 'page' | 'section' | 'card' | 'ui' | 'tokens'
  content: string
}

export interface GeneratedCode {
  files: GeneratedFile[]
  generatedAt: string
}

// 분석 실행
export interface AnalysisRun {
  id: string
  projectId: string
  status: AnalysisStatus
  designTokens?: DesignTokens
  aiInterpretation?: AiInterpretation
  generatedCode?: GeneratedCode
  errorMessage?: string
  createdAt: string
  completedAt?: string
}

export interface AnalysisStatusResponse {
  id: string
  status: AnalysisStatus
  currentStep?: string
  progress?: number
}

File: apps/api/schemas/analysis.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from pydantic.alias_generators import to_camel


class AnalysisRunRequest(BaseModel):
    """클라이언트 → 서버: camelCase 입력 허용"""
    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
    )

    project_id: uuid.UUID
    figma_json: dict


class AnalysisRunCreated(BaseModel):
    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
        from_attributes=True,
    )

    id: uuid.UUID
    status: str
    created_at: datetime


class AnalysisStatusResponse(BaseModel):
    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
        from_attributes=True,
    )

    id: uuid.UUID
    status: str
    current_step: str | None = None
    progress: int | None = None


class AnalysisRunResponse(BaseModel):
    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
        from_attributes=True,
    )

    id: uuid.UUID
    project_id: uuid.UUID
    status: str
    design_tokens: dict | None = None
    ai_interpretation: dict | None = None
    generated_code: dict | None = None
    error_message: str | None = None
    created_at: datetime
    completed_at: datetime | None = None
